### PR TITLE
✨ Registers `OSPARC_VARIABLE_API_*` as osparc session variable (🚨)

### PR DESCRIPTION
**Pull request by pcrespov**
<!-- Title Annotations:

  WIP: work in progress
  🐛    Fix a bug.
  ✨    Introduce new features.
  🎨    Enhance existing feature.
  ♻️    Refactor code.
  🚑️    Critical hotfix.
  ⚗️    Perform experiments.
  ⬆️    Upgrade dependencies.
  📝    Add or update documentation.
  🔨    Add or update development scripts.
  🔒️    Fix security issues.
  ⚠️    Changes in ops configuration etc. are required before deploying.
        [ Please add a link to the associated ops-issue or PR, such as in https://github.com/ITISFoundation/osparc-ops-environments or https://git.speag.com/oSparc/osparc-infra ]
  🗃️    Database table changed (relevant for devops).
  👽️    Public API changes (meaning: dev features are moved to being exposed in production)
  🚨    Do manual testing when deployed

or from https://gitmoji.dev/
-->

## What do these changes do?

Any service spawned by the `director-v2` (both computational and dynamic) will now have access to `OSPARC_VARIABLE_API_*` session variables auto-generated per user and product. Note that this api-key will **not** be renewed everytime until https://github.com/ITISFoundation/osparc-simcore/issues/5875 is in place. In the meantime, the user has the possibility to delete it in the UI.


- ♻️ Refactors osparc variables code base and moves into `service_service_director_v2.models.osparc_variables` submodule
    - ✨ All `OSPARC_VARIABLE_API_*` are now session variables created automatically **per user and product**
    - **This temporary lowers security** (see comments by @GitHK below)
    - Follow up on https://github.com/ITISFoundation/osparc-simcore/issues/5875
- ✨ `director-v2` CLI now lists osparc session variables
```cmd
$ cd services/director-v2
$ make install-dev
simcore-service osparc-variables --help
                                                                                                                                                                                                                                                                                         
 Usage: simcore-service osparc-variables [OPTIONS]                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                         
 Lists all registered osparc session variables                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                         
╭─ Options ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --help          Show this message and exit.                                        

$ simcore-service osparc-variables
OSPARC_VARIABLE_API_HOST
OSPARC_VARIABLE_API_KEY
OSPARC_VARIABLE_API_SECRET
OSPARC_VARIABLE_NODE_ID
OSPARC_VARIABLE_PRODUCT_NAME
OSPARC_VARIABLE_STUDY_UUID
OSPARC_VARIABLE_USER_EMAIL
OSPARC_VARIABLE_USER_ID
OSPARC_VARIABLE_USER_ROLE
```

## Related issue/s

- Resolves https://github.com/ITISFoundation/osparc-simcore/issues/5853
- Resolves https://github.com/ITISFoundation/osparc-simcore/issues/5397

## How to test

Driving tests `services/director-v2/tests/unit/test_utils_osparc_variables.py` and `services/director-v2/tests/unit/test_modules_api_keys_manager.py`

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by wvangeit — APPROVED** [review]
Looks good, thank you @pcrespov

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py [hunk 1]
```diff
@@ -98,7 +98,6 @@ async def create_user_services(app: FastAPI, scheduler_data: SchedulerData):
         node_id=scheduler_data.node_uuid,
         simcore_user_agent=scheduler_data.request_simcore_user_agent,
         swarm_stack_name=dynamic_services_scheduler_settings.SWARM_STACK_NAME,
-        run_id=scheduler_data.run_id,
     )
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py:101]: waow. you are disasembling? ;)

**Review by sanderegg — APPROVED** [review]
looking good. Am I correct when I understand that you removed the service lifetime entries? otherwise I don't quite get how the API secret/key would enter the computational services.
Then later we can probably definitely fix the 5Gb issue by actively using the osparc API when the data upload happens!

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks! looks good

**Comment by mrnicegyu11** [comment]
ack! thanks

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py:101]: I am confused by this, why did this not appear in the original function?

File: services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py [hunk 2]
```diff
@@ -71,97 +81,138 @@ async def is_used(self, identifier: str, cleanup_context: CleanupContext) -> boo
...
     *,
     product_name: ProductName,
     user_id: UserID,
     node_id: NodeID,
     run_id: RunID,
 ) -> ApiKeyGet:
-    api_keys_manager = _get_api_keys_manager(app)
-    display_name = _get_api_key_name(node_id, run_id)
+
+    warnings.warn(
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py:126]: waht is this for?

File: services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py [hunk 3]
```diff
@@ -69,6 +70,73 @@ async def substitute_vendor_secrets_in_model(
...
+            table.register_from_context(name, context_name)
+
+        table.register_from_handler("OSPARC_VARIABLE_USER_EMAIL")(request_user_email)
+        table.register_from_handler("OSPARC_VARIABLE_USER_ROLE")(request_user_role)
+        table.register_from_handler("OSPARC_VARIABLE_API_KEY")(
+            get_or_create_user_api_key
+        )
+        table.register_from_handler("OSPARC_VARIABLE_API_SECRET")(
+            get_or_create_user_api_secret
+        )
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py:131]: I have one issue with this. It lowers securtity. I did not see any mention of this in the body of the PR.
Could you please trace it in an issue.
Security goes down becase: all services will now start with the same key. A key will not be created and removed each time the service is started and shut down.

FYI: @elisabettai @mguidon @sanderegg

File: services/director-v2/tests/unit/test_utils_osparc_variables.py [hunk 4]
```diff
@@ -223,8 +248,8 @@ async def test_substitute_vendor_secrets_in_specs(
     mock_get_vendor_secrets: None, fake_app: FastAPI, faker: Faker
 ):
     specs = {
-        "api_key": "${OSPARC_VARIABLE_VENDOR_SECRET_ONE}",
-        "api_secret": "${OSPARC_VARIABLE_VENDOR_SECRET_TWO}",
+        "vendor_secret_one": "${OSPARC_VARIABLE_VENDOR_SECRET_ONE}",
+        "vendor_secret_two": "${OSPARC_VARIABLE_VENDOR_SECRET_TWO}",
```
Comment (GitHK) [inline services/director-v2/tests/unit/test_utils_osparc_variables.py:252]: Q: why was this renamed?

**Review by GitHK — APPROVED** [review]

**Review by GitHK — APPROVED** [review]
Please consider my comments

File: services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py (hunk 3 above)
Comment (pcrespov) [reply services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py:131]: @GitHK As we agreed in the offline peer review, I will follow up on this in https://github.com/ITISFoundation/osparc-simcore/issues/5875 (as referred in the description)

File: services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py (hunk 2 above)
Comment (pcrespov) [reply services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py:126]: It's a deprecation warning. The new interface is in a different module, as described in the deprecation error

File: services/director-v2/tests/unit/test_utils_osparc_variables.py (hunk 4 above)
Comment (pcrespov) [reply services/director-v2/tests/unit/test_utils_osparc_variables.py:252]: AFAIK it does not have any effect. I just wanted to align the names ...

File: services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py (hunk 2 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py:126]: all good missed the message

**Review by GitHK — APPROVED** [review]
👍

**Review by bisgaard-itis — APPROVED** [review]
Very cool 🥇 
Thanks a lot Pedro
